pkg/controller/common/defaults/pod_template.go

```diff
@@ -300,3 +301,13 @@ func (b *PodTemplateBuilder) WithAutomountServiceAccountToken() *PodTemplateBuil
 	}
 	return b
 }
+
+func (b *PodTemplateBuilder) WithFsGroup(defaultFsGroup int64) *PodTemplateBuilder {
+	if b.PodTemplate.Spec.SecurityContext == nil {
+		b.PodTemplate.Spec.SecurityContext = &corev1.PodSecurityContext{}
+	}
+	if b.PodTemplate.Spec.SecurityContext.FSGroup == nil {
+		b.PodTemplate.Spec.SecurityContext.FSGroup = pointer.Int64(defaultFsGroup)
+	}
+	return b
+}
```

Comment:
I think we'd want an escape hatch from that default in the podTemplate. Users should be allowed to express: "I want to override my fsGroup to `nothing`".

It's probably not easy to do at the `fsGroup` level, since it's an int pointer.
However I was thinking we could do it at the `securityContext` level:

```yaml
apiVersion: elasticsearch.k8s.elastic.co/v1
kind: Elasticsearch
metadata:
  name: elasticsearch-sample
spec:
  version: 8.0.0-SNAPSHOT
  nodeSets:
  - name: default
    count: 1
    podTemplate:
      spec:
        securityContext: {}
```

When I specify the above, I would like ECK to not set the fsGroup (even though the operator has the flag enabled).

So the `WithFsGroup ` function behaviour would be:

- if the securityContext is nil: set a default one with the default fsGroup
- if the securityContext is not nil (could be empty): don't set the fsGroup, consider the user want their own securityContext applied.

Given this, I'm wondering whether we should name the flag `set-default-security-context` instead.

Reply:
We discussed this offline with @david-kow, the use cases for the above are not super convincing:

- the operator is misconfigured, maybe you cannot reconfigure it for some reasons (eg. installed from operatorhub, or you don't have access to the operator namespace), but you'd still want to escape the fsGroup problem
- there are multiple PSPs on the cluster, you may want fsGroup set for some resources in some namespaces, but not for other resources in other namespaces

I still think it's a nice-to-have escape (also probably easier for just running the quickstart on Openshift without altering the operator config). Curious about other people's thought though @elastic/cloud-k8s.

Reply:
I have a slight preference for Seb's suggestion as it seems to give users an additional way out for any case we have not considered and it seems also reasonable to default the whole security context instead of just a single attribute.